tests/test_callprofiler.py

```diff
@@ -380,4 +380,47 @@ def foo():
         modmesh.call_profiler.reset()
         foo()
 
+    def test_status(self):
+        modmesh.wrapper_profiler_status.disable()
+        self.assertFalse(modmesh.wrapper_profiler_status.enabled)
+
+        modmesh.call_profiler.reset()
+        buf = modmesh.ConcreteBuffer(10)
+        result = modmesh.call_profiler.result()
+        self.assertEqual({}, result)
+
+        modmesh.wrapper_profiler_status.enable()
+        modmesh.call_profiler.reset()
+        buf = modmesh.ConcreteBuffer(10)  # noqa: F841
+        result = modmesh.call_profiler.result().get("children")
+        names = [data["name"] for data in result]
+        self.assertEqual(['ConcreteBuffer.__init__'], names)
```

Comment:
Since TimeRegistry have `modmesh.time_registry.names` to extract all names of function, but CallProfiler doesn't. Therefore, I fetch the names in result and compare it.

```diff
-         self.assertEqual(
-             ['ConcreteBuffer.__init__'],
-             modmesh.time_registry.names)
+         result = modmesh.call_profiler.result().get("children")
+         names = [data["name"] for data in result]
+         self.assertEqual(['ConcreteBuffer.__init__'], names)